src/compiler/crystal/command.cr

```diff
@@ -224,6 +227,13 @@ class Crystal::Command
         {$?, Time.now - start_time}
       ensure
         File.delete(output_filename) rescue nil
+
+        # Delete related dwarf generated by dsymutil, if any exists
+        {% if flag?(:darwin) %}
+          if compiler.generates_dsymutil?
+            File.delete("#{output_filename}.dwarf") rescue nil
```

Comment:
Maybe deleting the file should happen even if `--no-dsymutil` is used.
So after some compilation in debug mode with default options, the next compilation in release (or debug with `--no-dsymutil`) won't have a misleading dwarf.

Reply:
`execute` is only invoked when you do `crystal foo.cr` and a temporary executable is created in the cache directory. So in this case I don't see a problem.

For the case where you do `crystal build foo.cr` and then `crystal build foo.cr --no-debug` I wouldn't delete the dwarf file because we don't know if we are the ones who generated it in a previous run. And in any case dwarf files have a UUID that relates them with the executable, so an old dwarf file won't cause problems.

Reply:
🙆‍♂️

Reply:
Note that the misleading `.dwarf` file won't be processed if it doesn't match the executable —they both have an UUID section that must be identical.